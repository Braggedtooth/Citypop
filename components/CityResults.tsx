import React from 'react'
import { View } from 'react-native'
import { Text, Card, useTheme } from '@rneui/themed'
import { CityResponse } from '../types/Response'
import { CityResultsProps } from '../types/Components'

const CityResults = ({ results }: CityResultsProps) => {
  const { theme } = useTheme()

  return (
    <Card containerStyle={{ borderRadius: 10, backgroundColor: theme.colors.searchBg }}>
      <Card.Title style={{ fontSize: 20 }} testID="title">
        Welcome to {results.name}
      </Card.Title>
      <View
        style={{
          alignItems: 'center',
          justifyContent: 'center',
          backgroundColor: theme.colors.black,
          padding: 10,
          borderRadius: 5
        }}
      >
        <Text
          testID="population"
          style={{
            fontSize: 40,
            color: theme.colors.success,
            fontWeight: 'bold'
          }}
        >
          {results.population.toLocaleString()}
        </Text>
      </View>
      <Text style={{ fontSize: 15, textAlign: 'center', marginTop: 5, color: theme.colors.greyOutline }}>
        People live here
      </Text>
    </Card>
  )
}

export default CityResults
