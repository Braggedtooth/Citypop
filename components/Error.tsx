import { View } from 'react-native'
import React from 'react'
import { Text, useTheme } from '@rneui/themed'
const Error = () => {
  const { theme } = useTheme()
  return (
    <View
      style={{
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: theme.colors.black,
        padding: 10,
        marginHorizontal: 20,
        marginVertical: 10,
        borderRadius: 5
      }}
    >
      <Text style={{ color: theme.colors.error, textAlign: 'center' }}>
        Woooah! Didnt expect that? Welp it happened.. This is an error
      </Text>
    </View>
  )
}

export default Error
