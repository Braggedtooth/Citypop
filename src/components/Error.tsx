import { View } from 'react-native'
import React from 'react'
import { Text, useTheme } from '@rneui/themed'
type errorProps = {
  empty?: boolean
  value?: string
}
const Error = ({ empty, value }: errorProps) => {
  const { theme } = useTheme()
  return (
    <View
      style={{
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: theme.colors.black,
        padding: 10,
        marginHorizontal: 20,
        marginVertical: 10,
        borderRadius: 5
      }}
    >
      <Text testID="error" style={{ color: !empty ? theme.colors.warning : theme.colors.error, textAlign: 'center' }}>
        {!empty
          ? `Woah there! We couldnt find ${value} check for typos and try again`
          : `Woooah! Didnt expect that? Welp it happened.. This is an error`}
      </Text>
    </View>
  )
}

export default Error
