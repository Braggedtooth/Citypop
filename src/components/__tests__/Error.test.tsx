import { render } from '@testing-library/react-native'
import React from 'react'
import CityResults from '../CityResults'
import Error from '../Error'

describe('Test City Results', () => {
  let props = { empty: true, value: '' }
  beforeEach(() => {
    props = {
      empty: false,
      value: 'Nice'
    }
  })
  const renderResult = () => render(<Error empty={props.empty} value={props.value} />)
  it('Renders Correctly', () => {
    props.empty = true
    const component = renderResult()
    const error = component.getByTestId('error')
    expect(error.props.children).toBe(`Woooah! Didnt expect that? Welp it happened.. This is an error`)
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('Should show not found value', () => {
    const component = renderResult()
    const error = component.getByTestId('error')
    expect(error.props.children).toBe(`Woah there! We couldnt find ${props.value} check for typos and try again`)
  })
})
