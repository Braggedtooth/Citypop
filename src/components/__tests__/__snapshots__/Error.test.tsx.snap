// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test City Results Renders Correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#242424",
      "borderRadius": 5,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "marginVertical": 10,
      "padding": 10,
    }
  }
>
  <Text
    accessibilityRole="text"
    style={
      Object {
        "color": "#ff190c",
        "textAlign": "center",
      }
    }
    testID="error"
  >
    Woooah! Didnt expect that? Welp it happened.. This is an error
  </Text>
</View>
`;
